Fails to import pureimage with moduleResolution: nodenext

Fails to import pureimage with moduleResolution: nodenext since there are many 

### Expected Behaviour

```
error TS2339: Property 'make' does not exist on type 'typeof import("/Users/anatawa12/IdeaProjects/pureimage-nodenext-module-resolution/node_modules/pureimage/dist/index", { with: { "resolution-mode": "import" } })'.
error TS2339: Property 'encodePNGToStream' does not exist on type 'typeof import("/Users/anatawa12/IdeaProjects/pureimage-nodenext-module-resolution/node_modules/pureimage/dist/index", { with: { "resolution-mode": "import" } })'.
```

Does not occurs

### Actual Behaviour

```
error TS2339: Property 'make' does not exist on type 'typeof import("/Users/anatawa12/IdeaProjects/pureimage-nodenext-module-resolution/node_modules/pureimage/dist/index", { with: { "resolution-mode": "import" } })'.
error TS2339: Property 'encodePNGToStream' does not exist on type 'typeof import("/Users/anatawa12/IdeaProjects/pureimage-nodenext-module-resolution/node_modules/pureimage/dist/index", { with: { "resolution-mode": "import" } })'.
```

occurs because there are many `./filename-without-dot-js` in `.d.ts` files

### Steps To Reproduce

Configure typescript to use `"moduleResolution": "nodenext"` and import `pureimage`

https://github.com/anatawa12/pureimage-nodenext-module-resolution is the minimum reproducible repository. run `npm run build` to run `tsc` 

### Any Relevant Code Snippets

<!-- Please try to separate code snippets out for clarity -->

```ts
import * as pureImage from 'pureimage';

const canvas = pureImage.make(128, 128);

pureImage.encodePNGToStream(canvas, process.stdout)
```

### Platform

**OS:**  macOS 14.4.1 23E224 arm64 
**Node Version:** v20.10.0
**NPM Version:** 10.2.3
**PureImage Version:** 0.4.13

### Any Additional Info

<!--
Use this section to record any additional info you think might be helpful(for example a proposed
solution). Screenshots are also appreciated if applicable, as well as a minimal example demonstrating
the issue.
-->

original problem: https://github.com/misskey-dev/misskey/pull/13624#issuecomment-2047301018

--- a/src/context.ts
+++ b/src/context.ts
@@ -1,11 +1,11 @@
 import { Line } from "./line.js";
-import { NAMED_COLORS, TRANSPARENT_BLACK } from "./named_colors";
-import { Bounds, calc_min_bounds, Point, PointIsh, toRad } from "./point";
-import * as TEXT from "./text";
-import { Font } from "./text";
-import * as G from "./gradients";
-import { and, fromBytesBigEndian, getBytesBigEndian, or } from "./uint32";
-import type { Bitmap } from "./bitmap";
+import { NAMED_COLORS, TRANSPARENT_BLACK } from "./named_colors.js";
+import { Bounds, calc_min_bounds, Point, PointIsh, toRad } from "./point.js";
+import * as TEXT from "./text.js";
+import { Font } from "./text.js";
+import * as G from "./gradients.js";
+import { and, fromBytesBigEndian, getBytesBigEndian, or } from "./uint32.js";
+import type { Bitmap } from "./bitmap.js";
 import {
   MinimumBounds,
   PATH_COMMAND,
@@ -14,8 +14,8 @@ import {
   TextAlign,
   TextBaseline,
 } from "./types.js";
-import { clamp, colorStringToUint32 } from "./util";
-import { Matrix, Transform } from "./transform";
+import { clamp, colorStringToUint32 } from "./util.js";
+import { Matrix, Transform } from "./transform.js";
 
 /**
  * Used for drawing rectangles, text, images and other objects onto the canvas element. It provides the 2D rendering context for a drawing surface.

--- a/src/index.ts
+++ b/src/index.ts
@@ -1,4 +1,4 @@
 export * from "./bitmap.js";
 export * from "./text.js";
-export * from "./context";
-export * from "./image";
+export * from "./context.js";
+export * from "./image.js";

